### DISCUSSION TITLE
Idea about Web integration/Articles analisys

### DISCUSSION THREAD

**Discussion by xor2003** [ggml-org/llama.cpp#470] [Ideas]
Is it possible to make bridge to web for something unknown to model? (ChatGPT introduced plugins to search web, etc)
Or at least for model to read article/book and answer questions about it? 

What I understood how ChatPDF works:

1. Indexing document like:
word -> page, line
2. Put index into conversation context
3. Ask GPT to search inside index.
4. Jump into target page and provide paragraph into conversation context.
Maybe several times.
5. Answer user.

So no need for GPT to read whole document.
Probably can be implemented with LLaMa

**Comment by Technetium1** [comment]
See https://github.com/ggerganov/llama.cpp/discussions/541

**Comment by xor2003** [comment]
I have change the theme and content so previous comment is not relevant anymore

**Comment by FNsi** [comment]
I think there maybe a way

add a flag and something can do bing, 
using the flag and a special mark in your input to do search, or receive response from api

e.g --online -p "what I need to do?($Do€on¥line£)"

--online is the flag, 

($Do€on¥line£) is the mark,

The workflow I think might be:

0.app function find the mark
1.app search the web for the whole input without the mark, 
2.store the web search/ api result as prompt(online)
3.Feed the prompt in background process, like "you suddenly know that + result prompt(online)"
4.send your no mark input prompt to llama.
5.now llama may give you the answer from internet.


I think that may make sense, especially because search engines are given the best result directly in these days. And the problems:
1.multi times slower while doing online base on the result length.

2.if try to let llama to solve some logical questions, the internet result still not a good answer, or need more work to let it do step by step.

Let's wait the code Heroes!

**Comment by Topping1** [comment]
> Is it possible to make bridge to web for something unknown to model? (ChatGPT introduced plugins to search web, etc) Or at least for model to read article/book and answer questions about it?
> 
> What I understood how ChatPDF works:
> 
>     1. Indexing document like:
>        word -> page, line
> 
>     2. Put index into conversation context
> 
>     3. Ask GPT to search inside index.
> 
>     4. Jump into target page and provide paragraph into conversation context.
>        Maybe several times.
> 
>     5. Answer user.
> 
> 
> So no need for GPT to read whole document. Probably can be implemented with LLaMa

In theory you could modify this app:
[https://huggingface.co/spaces/pszemraj/summarize-long-text](https://huggingface.co/spaces/pszemraj/summarize-long-text)
and then use [https://github.com/PotatoSpudowski/fastLLaMa](https://github.com/PotatoSpudowski/fastLLaMa) to replace the calls to the transformers library.
